boards/xtensa/kincony_kc868_a32/kincony_kc868_a32.dts

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright (c) 2023 Bartosz Bilas
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+/dts-v1/;
+
+#include <espressif/esp32/esp32_wroom_32ue_n4.dtsi>
+#include "kincony_kc868_a32-pinctrl.dtsi"
+
+/ {
+	model = "Kincony KC868-A32";
+	compatible = "espressif,esp32";
+
+	aliases {
+		uart-0 = &uart0;
+		watchdog0 = &wdt0;
+	};
+
+	chosen {
+		zephyr,sram = &sram0;
+		zephyr,console = &uart0;
+		zephyr,shell-uart = &uart0;
+		zephyr,flash = &flash0;
+		zephyr,code-partition = &slot0_partition;
+	};
+};
+
+&cpu0 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+	cpu-power-states = <&light_sleep &deep_sleep>;
+};
+
+&cpu1 {
+	clock-frequency = <ESP32_CLK_CPU_240M>;
+};
+
+&gpio0 {
+	status = "okay";
+};
+
+&gpio1 {
+	status = "okay";
+};
+
+&i2c0 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 15 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 13 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c0_default>;
+	pinctrl-names = "default";
+
+	pcf8574@21 {
+		 compatible = "nxp,pcf8574";
+		 reg = <0x21>;
+		 gpio-controller;
+		 #gpio-cells = <2>;
+		 ngpios = <8>;
+	};
+
+	pcf8574@22 {
+		 compatible = "nxp,pcf8574";
+		 reg = <0x22>;
+		 gpio-controller;
+		 #gpio-cells = <2>;
+		 ngpios = <8>;
+	};
+
+	pcf8574@24 {
+		 compatible = "nxp,pcf8574";
+		 reg = <0x24>;
+		 gpio-controller;
+		 #gpio-cells = <2>;
+		 ngpios = <8>;
+	};
+
+	pcf8574@25 {
+		 compatible = "nxp,pcf8574";
+		 reg = <0x25>;
+		 gpio-controller;
+		 #gpio-cells = <2>;
+		 ngpios = <8>;
+	};
+};
+
+&i2c1 {
+	status = "okay";
+	clock-frequency = <I2C_BITRATE_STANDARD>;
+	sda-gpios = <&gpio0 4 GPIO_OPEN_DRAIN>;
+	scl-gpios = <&gpio0 5 GPIO_OPEN_DRAIN>;
+	pinctrl-0 = <&i2c1_default>;
+	pinctrl-names = "default";
+
+	pcf8574@21 {
```

Comment:
Both I2C0 and I2C1 contains the same IO expander. Is it correct?

Reply:
There are 8 identical expanders on the board, 4 for inputs and 4 for outputs. They have the same i2c address but are connected to different buses.

Reply:
@sylvioalves I was wondering if I should give them unique node labels as they cannot be used using the i2c shell at the moment as the device list returns duplicate device IDs.

<pre>
uart:~$ device list
devices:
- pcf8574@25 (READY)
- pcf8574@24 (READY)
- pcf8574@22 (READY)
- pcf8574@21 (READY)
- pcf8574@25 (READY)
- pcf8574@24 (READY)
- pcf8574@22 (READY)
- pcf8574@21 (READY)
</pre>

We could also improve this by adding an i2c bus prefix to the identifier, for example:

<pre>
uart:~$ device list
devices:
- i2c0_pcf8574@25 (READY)
- i2c0_pcf8574@24 (READY)
- i2c0_pcf8574@22 (READY)
- i2c0_pcf8574@21 (READY)
- i2c1_pcf8574@25 (READY)
- i2c1_pcf8574@24 (READY)
- i2c1_pcf8574@22 (READY)
- i2c1_pcf8574@21 (READY)
</pre>

Reply:
That is interesting. I guess a new PR enabling such prefix is the way.

Reply:
This can be done in this PR, we just need to do something like this:

<pre>
    i2c0_pcf8574@21 {
         compatible = "nxp,pcf8574";
         reg = <0x21>;
         gpio-controller;
         #gpio-cells = <2>;
         ngpios = <8>;
    };
</pre>

Reply:
ahá, of course. When I read it before I though you wanted to add the peripheral prefix in every and all instances.  
Sure, it makes sense.

Reply:
Done.